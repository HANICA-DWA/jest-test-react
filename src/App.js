import React from 'react';
import { connect } from 'react-redux'

import TodoItem from './TodoItem'
import AddTodoItem from './AddTodoItem'
import { addTodo } from './fakereducer'
import './App.css';

export function App (props) {
  return (
    <div className="App">
      <AddTodoItem addTodo={(e) => props.addTodo(e)} />
      {props.todos.map(todo => <TodoItem key={todo.id} todo={todo} />)}
    </div>
  );
}

const mapStateToProps = (state) => {
  return {
    todos: state.todos
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    addTodo: (todo) => dispatch(addTodo(todo))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
