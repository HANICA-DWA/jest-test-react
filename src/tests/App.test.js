import React from 'react'
import { Provider } from 'react-redux'
import App, {App as Unconnected} from '../App'
import { renderer } from './testhelpers'


test('Unconnected App is rendered', () => {
  const comp = renderer.create(<Unconnected todos={[]}/>)
  expect(comp.toJSON()).toMatchSnapshot();
})

test('Connected App is rendered', () => {
  const comp = renderer.create(<App />)
  expect(comp.toJSON()).toMatchSnapshot();
})

