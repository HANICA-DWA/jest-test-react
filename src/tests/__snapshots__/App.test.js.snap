// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unconnected App is rendered 1`] = `
<div
  className="App"
>
  <form
    onSubmit={[Function]}
  >
    <input
      name="todo"
      placeholder="Type a todo and press Enter.."
    />
  </form>
</div>
`;
